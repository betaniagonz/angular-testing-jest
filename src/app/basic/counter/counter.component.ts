import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-counter',
  templateUrl: './counter.component.html'
})
export class CounterComponent {

  counter: number = 10;

  increaseBy(value: number): void {
    this.counter += value;
    //TODO:
    //console.log({ newValue: this.counter });
  }

}
