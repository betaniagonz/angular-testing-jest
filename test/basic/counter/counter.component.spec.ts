import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from '../../../src/app/basic/counter/counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compiled: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [CounterComponent]
    });
    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should match the snapshot', () => {
      expect( compiled ).toMatchSnapshot();
  });

  test('increaseBy should increment the argument', () => {
    component.increaseBy(5);
    expect(component.counter).toBe(15);
  });

  test('click buttons should increment and decrement 1', () => {
    const buttons = compiled.querySelectorAll('button');
    buttons[0].click();
    expect(component.counter).toBe(11);
    buttons[1].click();
    buttons[1].click();
    expect(component.counter).toBe(9);
  });

  test('change the counter should change the value in the template (h1)', () => {
    component.increaseBy(10);
    fixture.detectChanges();
    const h1 = compiled.querySelector('h1');
    expect(h1?.textContent).toContain('20');
  });
});
