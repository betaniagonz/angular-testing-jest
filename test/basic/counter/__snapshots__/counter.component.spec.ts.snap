// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent should match the snapshot 1`] = `
<div
  id="root1"
>
  <h1>
    Counter 10
  </h1>
  <hr />
  <button
    class="btn btn-primary mx-1"
  >
    +1
  </button>
  <button
    class="btn btn-primary mx-1"
  >
    -1
  </button>
</div>
`;

exports[`CounterComponent should match the snapshot 2`] = `
<div
  id="root2"
>
  <h1>
    Counter 10
  </h1>
  <hr />
  <button
    class="btn btn-primary mx-1"
  >
    +1
  </button>
  <button
    class="btn btn-primary mx-1"
  >
    -1
  </button>
</div>
`;
